### PR TITLE
feat: experiment service initial commit  

### PR DESCRIPTION
**Pull request by jgellin-sf**
### What does this PR do?
Initial commit of experiment service.

Intention: Consuming code uses `getExperimentService()` to obtain object implementing `IExperimentService`, but do not access implementation object directly.

### What issues does this PR fix or reference?

@W-16326689@

### CODE REVIEW

File: test/unit/experimentService.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ **/
+
+import * as vscode from 'vscode';
```
Comment (gbockus-sf) [inline test/unit/experimentService.test.ts:8]: Should I put the mock stuff for VSCode in place?

**Review by gbockus-sf — DISMISSED** [review]
🚢

**Comment by gbockus-sf** [comment]
@jgellin-sf Updated with latest which I think included the package-lock.json file.  Should now show up as changed locally for commit.

File: test/unit/experimentService.test.ts (hunk 1 above)
Comment (jgellin-sf) [reply test/unit/experimentService.test.ts:8]: I started doing that but it turned out I didn't need it for this test file so I abandoned it.

**Review by gbockus-sf — APPROVED** [review]
🚢

**Pull request merged**